### PR TITLE
Native heic conversion

### PR DESCRIPTION
**Pull request by abhinavkgrd**

### CODE REVIEW

File: src/services/heicConvertor.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            rmSync(tempOutputFilePath);
+        } catch (e) {
+            logErrorSentry(e, 'failed to remove tempOutputFile');
+        }
+    }
+}
+
+function generateRandomName(length: number) {
+    let result = '';
+    const characters =
```
Comment (ua741) [inline src/services/heicConvertor.ts:66]: this const can be outside?

File: src/utils/preload.ts [hunk 2]
```diff
@@ -14,3 +14,11 @@ export const fixHotReloadNext12 = () => {
     })
   });`);
 };
+
+export function isPlatformMac() {
+    return process.platform === 'darwin';
```
Comment (ua741) [inline src/utils/preload.ts:19]: curious: Is this method/logic recommended somewhere?

File: src/api/heicConvert.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { ipcRenderer } from 'electron/renderer';
+import { isPlatformMac } from '../utils/preload';
+
+export async function convertHEIC(fileData: Uint8Array): Promise<Uint8Array> {
+    if (!isPlatformMac()) {
+        throw Error('native heic conversion only supported on mac');
+    }
+    const convertedFileData = await ipcRenderer.invoke(
+        'convert-heic',
```
Comment (ua741) [inline src/api/heicConvert.ts:9]: what's the command which is eventually invoked?

File: src/utils/preload.ts (hunk 2 above)
Comment (abhinavkgrd) [reply src/utils/preload.ts:19]: https://nodejs.org/api/process.html#process_process_platform , this is nodejs and should provide accurate value ?

is there any problem ?

File: src/api/heicConvert.ts (hunk 3 above)
Comment (abhinavkgrd) [reply src/api/heicConvert.ts:9]: this calls is handled by
>   ipcMain.handle('convert-heic', (_, fileData) => {
        return convertHEIC(fileData);
    });


 https://github.com/ente-io/bhari-frame/pull/83/files#diff-b49c6a18e63d8d58063c264aa42c5f179fd55a9b7a9cfb5211379ed52264eec5R102

File: src/services/heicConvertor.ts (hunk 1 above)
Comment (ua741) [reply src/services/heicConvertor.ts:66]: If you can moving, just move this entire method to some other random util. rest looks good.

**Review by ua741 — APPROVED** [review]

**Review by ua741 — APPROVED** [review]

**Pull request merged**